toolkit/tools/internal/csvparser/csvparser.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parses the timestamps stored in csv files and print them
+// to the terminal at the end of the build.
+
+package csvparser
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+var (
+	timeArray [][]string // A 2-D array used to store all timestamps from csvs.
+	// A list of csv files to be parsed.
+	files = []string{"/imageconfigvalidator.csv", "/imagepkgfetcher.csv", "/imager.csv", "/roast.csv"}
+)
+
+// Reads a CSV file and appends line by line to array.
+func CSVToArray(filename string) {
+	file, err := os.Open(filename)
+
+	if err != nil {
+		fmt.Println("failed to open csv file")
+	}
+	defer file.Close()
+
+	fileScanner := bufio.NewScanner(file)
+
+	for fileScanner.Scan() {
+		timeArray = append(timeArray, strings.Split(fileScanner.Text(), ","))
+	}
+
+}
+
+// Take list of file paths, parse, and output log to terminal.
+func OutputCSVLog(parentDir string) {
+	var startTime time.Time
+	init_file, err := os.Stat(parentDir + "/init")
+
+	// Format each file to array format.
+	for _, file := range files {
+		CSVToArray(parentDir + file)
+	}
+
+	// Get the start and end time from the first timestamp entry.
+	// Start time will be the ModTime for "init" if it exists, otherwise will be the first csv entry.
+	if os.IsNotExist(err) {
+		fmt.Printf("start: %s\n", timeArray[0][4])
```

Comment:
We should avoid using "magic numbers" - except for the author of this code, no one will know what "4" means and why this particular value. One solution would be to convert these to well-named constants, so their names would explain their purpose.